arch/arc/core/isr_wrapper.S

```diff
@@ -202,6 +202,44 @@ From RIRQ:
  */
 
 SECTION_FUNC(TEXT, _isr_wrapper)
+#ifdef CONFIG_EXECUTION_BENCHMARKING
+	/* Temporary save R0 */
+	st r0, [sp]
+
+	/* Disable stack checking */
+#ifdef CONFIG_ARC_STACK_CHECKING
+#ifdef CONFIG_ARC_HAS_SECURE
+	lr r0, [_ARC_V2_SEC_STAT]
+	bclr r0, r0, _ARC_V2_SEC_STAT_SSC_BIT
+	sflag r0
+#else
+	lr r0, [_ARC_V2_STATUS32]
+	bclr r0, r0, _ARC_V2_STATUS32_SC_BIT
+	kflag r0
+#endif
+#endif
+
+	push_s r1
```

Comment:
We need to save `blink` as well as `bl` (branch & link) instruction will set `blink` with return address and so we'll lose its original content which we may want to keep as a part of original (pre-IRQ) context.